pkg/controllers/management/catalog/controller.go

```diff
@@ -32,9 +32,17 @@ func runRefresh(ctx context.Context, interval int, controller v3.CatalogControll
 				logrus.Error(err)
 				continue
 			}
+			projectCatalogs, err := m.GetProjectCatalogs()
+			if err != nil {
+				logrus.Error(err)
+				continue
+			}
```

Comment:
On second thoughts, I would just add a separate refresh thread for Project Catalogs. So that error fetching either should not affect sync for the other type.